docs/ruleset_functions/are_all_terminal_CHW_loops_purchased_cooling.md

```diff
@@ -0,0 +1,28 @@
+# are_all_terminal_CHW_loops_purchased_cooling  
+
+**Description:** Returns TRUE if the fluid loop associated with the cooling_from_loop associated with each terminal unit is purchased CHW. Returns FALSE if this is not the case.   
+
+**Inputs:**  
+- **B-RMR**: To evaluate if the fluid loop associated with the cooling_from_loop associated with each terminal unit is purchased CHW.   
+- **terminal_unit_id_list**: List of terminal units to assess.  
+
+**Returns:**  
+- **are_all_terminal_CHW_loops_purchased_cooling**: Returns TRUE if the fluid loop associated with the cooling_from_loop associated with each terminal unit is purchased CHW. Returns FALSE if this is not the case.    
+ 
+**Function Call:** None  
+
+## Logic:   
+- Set are_all_terminal_CHW_loops_purchased_cooling equal to TRUE: `are_all_terminal_CHW_loops_purchased_cooling = TRUE`  
+- Check if RMR is not modeled with external fluid source: `if RMR.RuleModelInstance.external_fluid_source != Null:`  
+    - For each external fluid source in RMR: `for external_fluid_source.id in RMR.RuleModelInstance.external_fluid_source:`  
+        - Check if external fluid source is CHW: `if external_fluid_source.id.type == "CHILLED_WATER":`    
+            - Add the associated loop to the purchased_cooling_loop_list_b list: `purchased_cooling_loop_list_b = purchased_cooling_loop_list_b.append (external_fluid_source.id)`   
+- Create an object associate with the cooling_from_loop associated with terminal_b: `terminal_fluid_loop_b = terminal_b.cooling_from_loop`  
```

Comment:
The function has to loop through all terminals in the terminal_unit_id_list, but I don't see where this happens.

Reply:
@supriyagoel all updates made :)